src/App.jsx

```diff
@@ -9,16 +9,40 @@ const Heading = () => <h1>Todo list</h1>
 const App = () => {
   const [todoList, setTodoList] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-  useEffect(() => {
-    new Promise((resolve, reject) => 
-      setTimeout(
-        () => resolve({data: { todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [] } }),
-      2000
-      )
-    ).then((result) => {
-      setTodoList(result.data.todoList);
+
+  const fetchData = async () => {
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+      }
+    };
+    const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+    try {
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+
+      const data = await response.json();
+
+      const todos = data.records.map((todo)=> {
+        const newTodo =  {
+          id: todo.id,
+          title: todo.fields.title
+        }
+        return newTodo;
+      })
+      setTodoList(todos);
       setIsLoading(false);
-    }).catch(error => console.log("error", {error}))
+    } catch (error) {
+      console.error("Fetch API error:", error.message);
```

Comment:
nit: if an error happens, the page is stuck in loading state. in the future could be good to either setLoading false within the catch as well, or just do it after the try/catch block

Reply:
thank you